### PR TITLE
Update to the SplashFragment

### PR DESCRIPTION
**Pull request by datcodeboy**
changes to the splash screen fragments

### CODE REVIEW

**Comment by kulloveth** [comment]
is this ready for review

**Comment by datcodeboy** [comment]
yes it is

File: app/src/main/res/layout/fragment_splash.xml [hunk 1]
```diff
@@ -11,6 +11,6 @@
         android:id="@+id/splash_iv"
         android:layout_width="match_parent"
         android:layout_height="match_parent"
-        app:srcCompat="@drawable/splash_screen" />
+        android:background="@drawable/splash_screen" />
```
Comment (kulloveth) [inline app/src/main/res/layout/fragment_splash.xml:14]: this is an imageView why not use the drawable as a src

File: app/src/main/java/com/wellnesscity/health/ui/intro/SplashFragment.kt [hunk 2]
```diff
@@ -1,18 +1,27 @@
...
 import dagger.hilt.android.AndroidEntryPoint
 
 @AndroidEntryPoint
 class SplashFragment : Fragment() {
     private var binding: FragmentSplashBinding? = null
-
+    private val handler = Handler()
+    private val runnable = Runnable {
+        requireView().findNavController()
+            .navigate(R.id.action_splashFragment_to_onboardingFragment)
```
Comment (kulloveth) [inline app/src/main/java/com/wellnesscity/health/ui/intro/SplashFragment.kt:23]: We are using safeArgs so the destination id should be in safeArg format

File: app/src/main/java/com/wellnesscity/health/ui/intro/ViewPagerFragment.kt [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+
+import android.os.Bundle
+import androidx.fragment.app.Fragment
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import com.wellnesscity.health.R
+
+
+class ViewPagerFragment : Fragment() {
```
Comment (kulloveth) [inline app/src/main/java/com/wellnesscity/health/ui/intro/ViewPagerFragment.kt:11]: no need for this class the onboardingFragment is there already

File: app/src/main/java/com/wellnesscity/health/ui/intro/ViewPagerAdapter.kt [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+import androidx.fragment.app.Fragment
+import androidx.fragment.app.FragmentManager
+import androidx.lifecycle.Lifecycle
+import androidx.viewpager2.adapter.FragmentStateAdapter
+
+class ViewPagerAdapter(
+    list: ArrayList<Fragment>,
+    fm: FragmentManager,
+    lifecycle: Lifecycle
+): FragmentStateAdapter(fm, lifecycle) {
```
Comment (kulloveth) [inline app/src/main/java/com/wellnesscity/health/ui/intro/ViewPagerAdapter.kt:13]: We are using views only we dont need more fragments for this onboarding

File: app/src/main/res/navigation/nav_graph.xml [hunk 5]
```diff
@@ -11,15 +11,23 @@
...
-            app:popUpTo="@id/splashFragment"
-            app:popUpToInclusive="true" />
+            app:destination="@id/firstScreen" />
     </fragment>
     <fragment
         android:id="@+id/onboardingFragment"
         android:name="com.wellnesscity.health.ui.intro.OnboardingFragment"
         android:label="fragment_onboarding"
         tools:layout="@layout/fragment_onboarding" />
+    <fragment
```
Comment (kulloveth) [inline app/src/main/res/navigation/nav_graph.xml:28]: this should go as well since we dont need it

**Review by kulloveth — CHANGES_REQUESTED** [review]

**Review by kulloveth — COMMENTED** [review]
You forget to change the ViewPagerAdapter to use RecyclerAdapter, Also the direction to Onboarding fragment is still not using SafeArgs. Quickly fix this so I can merge

**Pull request merged**